code/game/objects/structures/crates_lockers/closets/bodybag.dm

```diff
@@ -67,4 +67,26 @@
 	icon_state = "bluebodybag"
 	foldedbag_path = /obj/item/bodybag/bluespace
 	mob_storage_capacity = 15
-	max_mob_size = MOB_SIZE_LARGE
\ No newline at end of file
+	max_mob_size = MOB_SIZE_LARGE
+
+/obj/structure/closet/body_bag/bluespace/MouseDrop(over_object, src_location, over_location)
+	..()
+	if(over_object == usr && Adjacent(usr) && (in_range(src, usr) || usr.contents.Find(src)))
+		if(!ishuman(usr))
+			return 0
+		if(opened)
+			return 0
+		if(contents.len >= mob_storage_capacity / 2)
+			usr << "<span class='warning'>There are too many things inside of [src] to fold it up!</span>"
+			return 0
+		for(var/obj/item/bodybag/bluespace/B in src)
+			usr << "<span class='warning'>You can't recursively fold bluespace body bags!</span>" //Nice try
+			return 0
+		visible_message("<span class='notice'>[usr] folds up [src].</span>")
+		var/obj/item/bodybag/B = new foldedbag_path(get_turf(src))
```

Comment:
@KorPhaeron Generally we like to create item versions of things on the fly rather than hold it. see: Drones, Pipes, Walls, Floors, Chairs. Only if the item is likely to have something that can change about it that we want to persist.